doc/developer-documentation.md

```diff
@@ -17,105 +17,147 @@ In order to enable a new tree, we need to add an entry there.
+After adding a `trees` entry, we need to define build and test configurations for it. In the same [configuration](https://github.com/kernelci/kernelci-pipeline/blob/main/config/pipeline.yaml) file, `jobs` section is there to specify them. `ChromeOS` specific job definitions are located in [config/jobs-chromeos.yaml](https://github.com/kernelci/kernelci-pipeline/blob/main/config/jobs-chromeos.yaml) file. Depending upon the type of the job such as build or test job, different parameters are specified:
 
 For instance,
-```yaml
-build_configs:
-
-    kernelci_staging-mainline:
-        tree: kernelci
-        branch: 'staging-mainline'
-        variants:
-            gcc-10:
-                build_environment: gcc-10
-                architectures:
-                    x86_64:
-                        base_defconfig: 'x86_64_defconfig'
-                        filters:
-                            - regex: { defconfig: 'x86_64_defconfig' }
-```
-
-That's it! The tree is enabled now.
-All the jobs defined under `jobs` section of [config file](https://github.com/kernelci/kernelci-pipeline/blob/main/config/pipeline.yaml) would run on the newly added tree until specified otherwise.
-
-
-## Enabling a new test
-
-KernelCI currently runs several test suites. This manual is intended
-to provide documentation for developers on how to enable new tests.
-
-
-### Job definition
-The pipeline [configuration](https://github.com/kernelci/kernelci-pipeline/blob/main/config/pipeline.yaml) file has `jobs` section for defining various build and test jobs. The currently enabled tests are `kver`, `baseline`, `kunit`, `kselftest`, and `sleep` tests.
-`ChromeOS` specific job definitions are located in [config/jobs-chromeos.yaml](https://github.com/kernelci/kernelci-pipeline/blob/main/config/jobs-chromeos.yaml) file.
-
-To enable a new test, an entry needs to be added as follows:
-
 ```yaml
 jobs:
-  <job-name>:
+
+  <kbuild-job-name>:
     template: <job-template>
-    kind: <kind of job, either 'kbuild', 'job' or 'test'>
+    kind: kbuild
     image: <docker-image-name>
-    params: <parameters such as arch, compiler, defconfig>
-    rules: <job-rules>
+    params:
+      arch: <architecture>
+      compiler: <compiler_name>
+      cross_compile: <compiler_prefix>
+      dtbs_check: <dtb_check_enabled>
+      defconfig: <defconfig_name>
+      fragments:
+       - <fragment_name>
+    rules:
+      min_version:
+        version: <kernel_version>
+        patchlevel: <kernel_patch_level>
+      tree:
+        - <tree_name>
+        - <!tree_name2>
+
+  <test-job-name>:
+    template: <template_name>
+    kind: <kind of job, either 'job' or 'test'>
+    params:
+      nfsroot: <rootfs_url>
+      collections: <name_of_suite>
+      job_timeout: <timeout>
     kcidb_test_suite: <kcidb-mapping>
+    rules:
+      min_version:
+        version: <kernel_version>
+        patchlevel: <kernel_patch_level>
+      tree:
+        - <tree_name>
+        - <!tree_name2>
```

Comment:
In this example and other ones as well, some property markers use `-`(e.g. `<kbuild-job-name>`) whereas others use `_`(e.g. `<tree_name>`).
Please unify a pattern in the whole file.

Reply:
Unified now.

Reply:
This doesn't look like it's fixed yet. Maybe you haven't committed the change yet?

Reply:
Updated now.

Reply:
It's not fixed yet. See my recent 2 comments.

Reply:
I'd fixed only trees. Thank you for suggesting. 

I'll take extra care in naming things in yaml from now on.